### PR TITLE
iOS: Mobile only subscription

### PR DESCRIPTION
**Pull request by ivan-magda**
**Description**

- Adds mobile only subscription on iOS

### CODE REVIEW

**Comment by ivan-magda** [comment]
@XanderZhu please take a look at `shared` changes

File: shared/src/iosMain/kotlin/org/hyperskill/app/core/domain/model/SwiftyResult.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+sealed class SwiftyResult<out S, out F : Throwable> {
+    data class Success<out S, out F : Throwable>(val value: S) : SwiftyResult<S, F>()
+    data class Failure<out S, out F : Throwable>(val failure: F) : SwiftyResult<S, F>()
+
+    internal fun toKotlinResult(): Result<S> =
+        when (this) {
+            is Success -> Result.success(value)
+            is Failure -> Result.failure(failure)
+        }
```
Comment (XanderZhu) [inline shared/src/iosMain/kotlin/org/hyperskill/app/core/domain/model/SwiftyResult.kt:11]: It's better to extract this kind of functions from data classes. I suggest making it an extension function.

File: shared/src/commonMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package org.hyperskill.app.purchases.domain.model
+
+internal expect object PlatformProductIdentifiers {
+    val mobileOnlySubscription: String
```
Comment (XanderZhu) [inline shared/src/commonMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt:4]: Let's make it in the UPPER_CASE way, as all the consts

File: shared/src/iosMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package org.hyperskill.app.purchases.domain.model
+
+internal actual object PlatformProductIdentifiers {
+    actual val mobileOnlySubscription: String = "org.hyperskill.App.MobileOnly.Monthly"
```
Comment (XanderZhu) [inline shared/src/iosMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt:4]:
```suggestion
    actual const val MOBILE_ONLY_SUBSCRIPTION: String = "org.hyperskill.App.MobileOnly.Monthly"
```

File: shared/src/commonMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt (hunk 2 above)
Comment (XanderZhu) [inline shared/src/commonMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt:4]:
```suggestion
val MOBILE_ONLY_SUBSCRIPTION: String
```

File: shared/src/androidMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package org.hyperskill.app.purchases.domain.model
+
+internal actual object PlatformProductIdentifiers {
+    actual val mobileOnlySubscription: String = "premium_mobile"
```
Comment (XanderZhu) [inline shared/src/androidMain/kotlin/org/hyperskill/app/purchases/domain/model/PlatformProductIdentifiers.kt:4]:
```suggestion
    actual const val MOBILE_ONLY_SUBSCRIPTION: String = "premium_mobile"
```

**Review by XanderZhu — APPROVED** [review]

**Pull request merged**